### ISSUE TITLE
Problem training network with weight regularization or dropout layers in parallel with joblib

### ISSUE THREAD

**Issue by GitAnt** [fchollet/keras#2877]
I'm having some problems in training a network using l2 regularization (or dropout layer) and joblib for parallel execution of the some cross validation runs.
The script below should reproduce the issue
https://gist.github.com/GitAnt/10d3285ff308724acde3522a0750d51f
while the output can be found here
https://gist.github.com/GitAnt/d464d6ef1ea6153fb09868858af8b560
